source/class/qx/util/format/DateFormat.js

```diff
@@ -1207,10 +1216,28 @@ qx.Class.define("qx.util.format.DateFormat",
         dateValues.hour = parseInt(value, 10) % 12;
       };
 
-      var ignoreManipulator = function(dateValues, value) {
-        return;
+      var timezoneManipulator = function(dateValues, value) {
+        var regEx = new RegExp("([+-]*)(\\d\\d)[:]*(\\d\\d)*$");
```

Comment:
This regex doesn't look right. It allows 0 or more leading `+` or `-`, and zero or more colons. That means that this value is accepted (but `tzResults[1]` does not equal `-`):
```
"---23:::13"
```

I think you probably mean "options" (0 or 1) in both cases, so you'd use this:
```
var regEx = new RegExp("([+-]?)(\\d\\d)[:]?(\\d\\d)?$");
```

Even that, though, doesn't invalidate bad input. I think there are additional constraints necessary to detect bad input.

Reply:
I think `var regEx = new RegExp("([+-]?)(\\d\\d)(?::?(\\d\\d))?$");` should do.

Reply:
Thanks, I've updated the regex. This regex is only used for extracting the parts, not for validation. Validation is already done via regexes on lines 1784 & 1791 for rules 'Z' and 'z' respectively.